100of100Service/src/test/java/dev/coms4156/project/RouteControllerTest.java

```diff
@@ -33,6 +33,7 @@ public class RouteControllerTest {
   private static final String CLIENT_ID_2 = "Mg";
   private static final String CLIENT_ID_99 = "OTk";
 
+  private static DatabaseConnection dbConnection;
   private static DatabaseConnectionStub dbConnectionStub;
```

Comment:
Err, I found a bug in our previous code: we do not need the `dbConnectionStub` variable but only your new variable.

Given that `DatabaseConnectionStub` is a sub-class of `DatabaseConnection`, we should actually cast the `dbConnectionStub` to the type of `DatabaseConnection` and provide to the Facade.

Nevertheless, you code is good, we can keep it for now, and refactor later.